### PR TITLE
Implement keyboardDidHide listener and open modal if the Keyboard appears

### PR DESCRIPTION
**Pull request by OlimpiaZurek**
Original issue: $https://github.com/Expensify/App/issues/16353

This PR changes implementation on Keyboard dismiss inside `togglePicker` method.  In current implementation `Keyboard.dismiss()` is called immediately, which can caused issues ( eg.  https://github.com/Expensify/App/issues/16353). 
I added listener `keyboardDidHide` and open modal if the Keyboard appears. Otherwise just open modal.

### CODE REVIEW

**Comment by 0xmiroslav** [comment]
@danieldoglas when user currently focuses on input and select picker and close picker, should we focus back to original input?

**Comment by danieldoglas** [comment]
@0xmiroslav I don't think we should go back to the original input, we should just close.

I'm addressing internally why I can't assign you to this PR, I think we have some permission issues here. Will get back to you soon.

**Comment by danieldoglas** [comment]
@0xmiroslav can you try to review this PR without me assigning you?

**Comment by 0xmiroslav** [comment]
> @0xmiroslav can you try to review this PR without me assigning you?

Sure thing

File: src/index.js [hunk 1]
```diff
@@ -292,6 +280,27 @@ export default class RNPickerSelect extends PureComponent {
...
+
+        if (disabled) {
+            return;
+        }
+
+        this.triggerOpenCloseCallbacks();
+
+        if (Keyboard.isVisible()) {
+            const keyboardListener = Keyboard.addListener('keyboardDidHide', () => {
+               this.updatePickerState(animate,postToggleCallback);
```
Comment (0xmiroslav) [inline src/index.js]:
```suggestion
               this.updatePickerState(animate, postToggleCallback);
```

File: src/index.js [hunk 2]
```diff
@@ -292,6 +280,27 @@ export default class RNPickerSelect extends PureComponent {
...
+               keyboardListener.remove();
+            });
+            Keyboard.dismiss();
+        }
+        else {
+            this.updatePickerState(animate,postToggleCallback);
```
Comment (0xmiroslav) [inline src/index.js]:
```suggestion
            this.updatePickerState(animate, postToggleCallback);
```

File: src/index.js [hunk 3]
```diff
@@ -292,6 +280,27 @@ export default class RNPickerSelect extends PureComponent {
...
+        this.triggerOpenCloseCallbacks();
+
+        if (Keyboard.isVisible()) {
+            const keyboardListener = Keyboard.addListener('keyboardDidHide', () => {
+               this.updatePickerState(animate,postToggleCallback);
+               keyboardListener.remove();
+            });
+            Keyboard.dismiss();
+        }
+        else {
```
Comment (0xmiroslav) [inline src/index.js]:
```suggestion
        } else {
```

**Review by 0xmiroslav — CHANGES_REQUESTED** [review]
Code mostly looks good. Just minor feedback:

**Comment by OlimpiaZurek** [comment]
all comments have already been resolved

**Comment by 0xmiroslav** [comment]
@OlimpiaZurek please pull from master and fix lint. Not related to this PR but asking since you also worked on https://github.com/Expensify/react-native-picker-select/pull/9.

```diff
- const preserveSpaces = (label) =>  {
+ const preserveSpaces = (label) => {
    return label.replace(/ /g, '\u00a0');
-  }
+ }
```

And it would be good to add some automated tests.

**Comment by OlimpiaZurek** [comment]
> @OlimpiaZurek please pull from master and fix lint. Not related to this PR but asking since you also worked on #9.
> 
> ```diff
> - const preserveSpaces = (label) =>  {
> + const preserveSpaces = (label) => {
>     return label.replace(/ /g, '\u00a0');
> -  }
> + }
> ```
> 
> And it would be good to add some automated tests.

done

**Review by 0xmiroslav — APPROVED** [review]
Looks good, tests well, automated tests pass.
Checklist will be done in app PR.
cc: @danieldoglas

**Review by danieldoglas — APPROVED** [review]

**Pull request merged**